packages/jellyfish-api-core/__tests__/category/masternode/updatemasternode.test.ts

```diff
@@ -0,0 +1,576 @@
+      await expect(promise).rejects.toThrow(`RpcApiError: 'operatorAddress (${invalidAddress}) does not refer to a P2PKH or P2WPKH address', code: -8, method: updatemasternode`)
+    }
+
+    {
+      const invalidAddress = 'INVALID_ADDRESS'
+      const promise = client.masternode.updateMasternode(masternodeId, {
+        rewardAddress: invalidAddress
+      })
+
+      await expect(promise).rejects.toThrow(RpcApiError)
+      await expect(promise).rejects.toThrow(`RpcApiError: 'rewardAddress (${invalidAddress}) does not refer to a P2PKH or P2WPKH address', code: -8, method: updatemasternode`)
+    }
+  })
+
+  it('should not update masternode while in PRE_ENABLED or TRANSFERRING state', async () => {
+    const initialAddress = await client.wallet.getNewAddress()
+    const masternodeId = await client.masternode.createMasternode(initialAddress)
+
+    await container.generate(1)
+
+    // test call before masternode active
+    const ownerAddress = await client.wallet.getNewAddress()
+    {
+      const promise = client.masternode.updateMasternode(masternodeId, { ownerAddress: ownerAddress })
+      await expect(promise).rejects.toThrow(RpcApiError)
+      await expect(promise).rejects.toThrow(`RpcApiError: 'Test UpdateMasternodeTx execution failed:\nMasternode ${masternodeId} is not in 'ENABLED' state', code: -32600, method: updatemasternode`)
+    }
+    {
+      const promise = client.masternode.updateMasternode(masternodeId, { operatorAddress: ownerAddress })
+      await expect(promise).rejects.toThrow(RpcApiError)
+      await expect(promise).rejects.toThrow(`RpcApiError: 'Test UpdateMasternodeTx execution failed:\nMasternode ${masternodeId} is not in 'ENABLED' state', code: -32600, method: updatemasternode`)
+    }
+    {
+      const promise = client.masternode.updateMasternode(masternodeId, { rewardAddress: ownerAddress })
+      await expect(promise).rejects.toThrow(RpcApiError)
+      await expect(promise).rejects.toThrow(`RpcApiError: 'Test UpdateMasternodeTx execution failed:\nMasternode ${masternodeId} is not in 'ENABLED' state', code: -32600, method: updatemasternode`)
+    }
+
+    {
+      const masternodes = await client.masternode.listMasternodes()
+      expect(masternodes[masternodeId]).toBeTruthy()
+      expect(masternodes[masternodeId].state).toStrictEqual('PRE_ENABLED')
+      expect(masternodes[masternodeId].ownerAuthAddress).toStrictEqual(initialAddress)
+    }
+
+    await container.generate(20)
+
+    // Check new state is TRANSFERRING and owner is still the same
+    await client.masternode.updateMasternode(masternodeId, {
+      ownerAddress: ownerAddress
+    })
+
+    await container.generate(20)
+
+    {
+      const masternodes = await client.masternode.listMasternodes()
+      expect(masternodes[masternodeId]).toBeTruthy()
+      expect(masternodes[masternodeId].state).toStrictEqual('TRANSFERRING')
+      expect(masternodes[masternodeId].ownerAuthAddress).toStrictEqual(initialAddress)
```

Comment:
```suggestion
      expect(data).toStrictEqual(expect.objectContaining({
        state: 'TRANSFERRING',
        ownerAuthAddress: initialAddress
      }))
```

For better readability, it's better to write tests like this. Most of the tests written previously were written the previous "style" that we've since discouraged. We don't have to change it now but just a thought for the future.